--- "c:\\Users\\mgt\\.platformio\\packages\\framework-arduinoadafruitnrf52\\cores\\nRF5\\linker\\nrf52840_s140_v6.ld"	2024-05-20 22:50:13.739714300 +0800
+++ "c:\\Users\\mgt\\Desktop\\proj\\gps_tracker\\patches\\src\\nrf52840_s140_v6.ld"	2025-05-06 23:26:43.119141100 +0800
@@ -5,7 +5,7 @@
 
 MEMORY
 {
-  FLASH (rx)     : ORIGIN = 0x26000, LENGTH = 0xED000 - 0x26000
+  FLASH (rx)     : ORIGIN = 0x26000, LENGTH = 0x48000 - 0x26000
 
   /* SRAM required by Softdevice depend on
    * - Attribute Table Size (Number of Services and Characteristics)
